src/main/java/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -114,4 +134,41 @@ public static String evalXpathExpression(XPathExpression expr,
         return new Pair<>(ExternalDataInstance.buildFromRemote(queryDatum.getDataId(), root, queryDatum.useCaseTemplate()), "");
     }
 
+    public void populateItemSetChoices(QueryPrompt queryPrompt) {
+        evaluationContext.setVariables(userAnswers);
+        ItemsetBinding itemset = queryPrompt.getItemsetBinding();
+        if (itemset != null) {
```

Comment:
Am I right in assuming this is all pulled ~wholesale from the Form Engine?

Am wondering whether it's identical enough to try to put this into a helper method of some kind. Some of the optimizations around populating Select Choices are pretty powerful, but would be very brittle to replicate